openc3/lib/openc3/utilities/store_autoload.rb

```diff
@@ -32,6 +33,29 @@
 end
 
 module OpenC3
+  class StoreConnectionPool < ConnectionPool
+    NO_OPTIONS = {}
+
+    def pipelined
+      with(NO_OPTIONS) do |redis|
+        redis.pipelined do |pipeline|
+          Thread.current[:pipeline] = pipeline
+          yield
+          Thread.current[:pipeline] = nil
+        end
+      end
+    end
+
+    def with(options = NO_OPTIONS, &block)
+      pipeline = Thread.current[:pipeline]
+      if pipeline
+        yield pipeline
+      else
+        super(options, &block)
+      end
+    end
```

Comment:
Does this basically allow you to call pipelined multiple times and it will reuse the current thread pipeline? Makes it re-entrant?

Reply:
It makes pipelining transparent.   We weren't doing it correctly before and it was effectively having no effect.  You have to make calls on the pipeline object rather than the normal redis client object.